cpp/src/io/comp/uncomp.cpp

```diff
@@ -422,17 +423,17 @@ std::vector<char> io_uncompress_single_h2d(const void* src, size_t src_size, int
  * @return Vector containing the output uncompressed data
  */
 std::vector<char> get_uncompressed_data(host_span<char const> const data,
-                                        std::string const& compression)
+                                        compression_type compression)
 {
   int comp_type = IO_UNCOMP_STREAM_TYPE_INFER;
-  if (compression == "gzip")
-    comp_type = IO_UNCOMP_STREAM_TYPE_GZIP;
-  else if (compression == "zip")
-    comp_type = IO_UNCOMP_STREAM_TYPE_ZIP;
-  else if (compression == "bz2")
-    comp_type = IO_UNCOMP_STREAM_TYPE_BZIP2;
-  else if (compression == "xz")
-    comp_type = IO_UNCOMP_STREAM_TYPE_XZ;
+
+  switch (compression) {
+    case compression_type::GZIP: comp_type = IO_UNCOMP_STREAM_TYPE_GZIP; break;
+    case compression_type::ZIP: comp_type = IO_UNCOMP_STREAM_TYPE_ZIP; break;
+    case compression_type::BZIP2: comp_type = IO_UNCOMP_STREAM_TYPE_BZIP2; break;
+    case compression_type::XZ: comp_type = IO_UNCOMP_STREAM_TYPE_XZ; break;
+    default: break;
+  }
```

Comment:
I think what you have is fine, but I would think this could optimize better:
```
int const comp_type = [&]() {
  switch (compression) {
    case compression_type::GZIP: return IO_UNCOMP_STREAM_TYPE_GZIP;
    case compression_type::ZIP: return IO_UNCOMP_STREAM_TYPE_ZIP;
    case compression_type::BZIP2: return IO_UNCOMP_STREAM_TYPE_BZIP2;
    case compression_type::XZ: return IO_UNCOMP_STREAM_TYPE_XZ;
    default: return IO_UNCOMP_STREAM_TYPE_INFER;
  }}();
```